component/src/main/java/org/wso2/extension/siddhi/io/ibmmq/sink/IBMMQSink.java

```diff
@@ -0,0 +1,238 @@
+            connectionFactory.setTransportType(WMQConstants.WMQ_CM_CLIENT);
+            connectionFactory.setPort(Integer.parseInt(optionHolder.validateAndGetOption(IBMMQConstants.PORT)
+                    .getValue()));
+            connectionFactory.setHostName(optionHolder.validateAndGetOption(IBMMQConstants.HOST).getValue());
+            connectionFactory.setChannel(optionHolder.validateAndGetOption(IBMMQConstants.CHANNEL).getValue());
+        } catch (JMSException e) {
+            throw new IBMMQSinkAdaptorRuntimeException("Error while initializing IBM MQ sink: " + optionHolder.
+                    validateAndGetOption(IBMMQConstants.DESTINATION_NAME).getValue() +
+                    ", " + e.getMessage(), e);
+        }
+
+    }
+
+    @Override
+    public void publish(Object payload, DynamicOptions transportOptions) throws ConnectionUnavailableException {
+        try {
+            messageSender = session.createSender(queue);
+            if (payload instanceof String) {
+                Message message = session.createTextMessage(payload.toString());
+                messageSender.send(message);
+
+            } else if (payload instanceof Map) {
+                MapMessage mapMessage = session.createMapMessage();
+                ((Map) payload).forEach((key, value) -> {
+                    try {
+                        mapMessage.setString((String) key, (String) value);
+                    } catch (JMSException e) {
+                        throw new IBMMQSinkAdaptorRuntimeException("Exception has occurred while publishing payload: " +
+                                "key - '" + key + "', value - '" + value + "' from stream: '"
+                                + outputStreamDefinition.getId() + "'. ", e);
+                    }
+                });
+                messageSender.send(mapMessage);
+            } else if (payload instanceof ByteBuffer) {
+                byte[] data = ((ByteBuffer) payload).array();
+                BytesMessage bytesMessage = session.createBytesMessage();
+                bytesMessage.writeBytes(data);
+                messageSender.send(bytesMessage);
+            }
+            messageSender.close();
+        } catch (JMSException e) {
+            throw new IBMMQSinkAdaptorRuntimeException("Exception has occurred while publishing payload: " +
+                    payload.toString() + " , ", e);
+        }
+    }
+
+    @Override
+    public void connect() throws ConnectionUnavailableException {
+        try {
+            if (isSecured) {
+                connection = (QueueConnection) connectionFactory.createConnection(userName, password);
+            } else {
+                connection = (QueueConnection) connectionFactory.createConnection();
+            }
+            session = (QueueSession) connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+            queue = session.createQueue(optionHolder.validateAndGetOption(IBMMQConstants.DESTINATION_NAME)
```

Comment:
Shouldn't we support dynamic destination names?
https://github.com/wso2-extensions/siddhi-io-jms/blob/master/component/src/main/java/org/wso2/extension/siddhi/io/jms/sink/JMSSink.java#L56

Reply:
We may not able to add that into this release, but if needed we can add it later and release a version.

Reply:
Done